woocommerce/woocommerce-ios
# [REST API] Migrate MediaRemote endpoints

Part of: #8566 


## Description

The requests from [MediaRemote](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Networking/Networking/Remote/MediaRemote.swift) are all WPCOM requests, and those requests aren't using Jetpack tunnel. 

In this PR, 
- We migrate the `DotcomRequest`s and make them available as `RESTRequest`s
- Check for placeholder site ID and use WordPressSite API. (Similar to [JCP checks](https://github.com/woocommerce/woocommerce-ios/pull/8686/files#diff-c1799bdb79b6ad1e6d338c866d29c8aabb85af9967581edd24646eb1549f4ad5R72))
- Add unit tests. 

## Testing instructions
- Login into a Jetpack-connected store.
- Navigate to the Products tab
- Tap `+` to create a new Product and add product images to it. 
- Upload from the device and validate that image upload works as expected.
- Try choosing images from "WordPress Media library" for a product. (Ref screenshot)
- Everything related to Media should work as before.
- Repeat the testing for a non-Jetpack JCP store. (Only WooCommerce payments and no Jetpack plugin)

RESTRequest migration can be tested from this PR - https://github.com/woocommerce/woocommerce-ios/pull/8687

## Screenshots
<img src="https://user-images.githubusercontent.com/524475/213327535-1fa125f3-629f-477f-b4d9-a46bfce627bc.png" width="320"/>


---
